lib/Parse/ParseDecl.cpp

```diff
@@ -2829,6 +2828,32 @@ void Parser::ParseDeclarationSpecifiers(DeclSpec &DS,
       AttrsLastTime = true;
       continue;
 
+    case tok::kw__Checked:
+      // Checked C - it is parsed as checked array or checked function keyword
+      // checked array type, checked []
+      if (NextToken().is(tok::l_square)) {
+        if (!getLangOpts().CPlusPlus11 || !isCXX11AttributeSpecifier())
```

Comment:
The "checked" keyword is only recognized when the compiler is parsing C.  This means that !getLangOpts().CPlusPlus11() will always be true.  I suggest simplifying this block of code.  I think the code beneath the goto can be deleted, and the if-statement can be removed.